src/plugins/data_views/common/data_views/data_view.ts

```diff
@@ -582,8 +583,18 @@ export class DataView implements DataViewBase {
    * Return the "runtime_mappings" section of the ES search query.
    */
   getRuntimeMappings(): estypes.MappingRuntimeFields {
-    // @ts-expect-error The ES client does not yet include the "composite" runtime type
-    return _.cloneDeep(this.runtimeFieldMap);
+    const mappedFields = this.getMappedFieldNames();
+    return Object.keys(this.runtimeFieldMap).reduce((acc, fieldName) => {
+      // do not include fields that are mapped
+      if (mappedFields.includes(fieldName)) {
+        return acc;
+      }
+
+      return {
```

Comment:
This object could be modified instead up duplicated but I'll leave this to up to you.

Reply:
766682de3c7